src/opengb/dram/quantity.cljc

```diff
@@ -28,7 +29,26 @@
          volume-intensity-unit?
          per-year-unit?))
 
-(s/def ::magnitude number?)
+(def us-customary-unit? #{"ft**2"
+                          "lb/year"
+                          "t/year"
+                          "kBtu/ft**2/year"
+                          "kWh/ft**2/year"
+                          "t/ft**2/year"
+                          "lb/ft**2/year"})
+
+(def metric-unit? #{"m**2"
+                    "kg/m**2/year"
+                    "t/m**2/year"
+                    "GJ/m**2/year"
+                    "kg/year"
+                    "Mg/year"
+                    "kWh/m**2/year"
+                    "l/m**2/year"
+                    "kWh/year"
+                    "l/year"})
+
+(s/def ::magnitude opengb.dram.util/strictly-a-number?)
```

Comment:
I'm not 100% sure this new constraint belongs here. Will it cause problems for us in the future to be excluding `##NaN` like this?

Reply:
Took another look at this ... would just leaning on the stock `(s/double-in :infinite? false :NaN? false)` spec do what we want? https://clojuredocs.org/clojure.spec.alpha/double-in

Reply:
Right, that sounds like a better idea.
But still, is there any case for creating quantities that contain `infinity` or `NaN`? Or can we safely disallow those special values?

Reply:
They’re physical quantities in the end so I can’t imagine a circumstance when we’d want to let them in. (Then again I’m not a cosmologist. But they can write their own library).